bot.py

```diff
@@ -38,25 +41,43 @@ def in_script_dir(file_path):
 class Bot:
     """Slack bot used to manage the release"""
 
-    def __init__(self, slack_webhook_url, access_token, repo_dir, version, rc_hash_url, prod_hash_url):
+    def __init__(self, websocket, access_token, channel_mapping):
         """
         Create the slack bot
 
         Args:
-            slack_webhook_url (str): A Slack webhook URL used to post in a Slack channel
+            websocket (obj): websocket for sending messages
             access_token (str): The OAuth access token used to interact with Slack
-            repo_dir (str): The directory of a git repository which will be cloned and used for the release
-            version (str): The version of the release
-            rc_hash_url (str): The URL used to poll the RC server to confirm deployment of the release candidate
-            prod_hash_url (str): The URL used to poll the production server to confirm deployment of production
+            channel_mapping (dict): Details for repos by channel_id
         """
-        self.slack_webhook_url = slack_webhook_url
+        self.websocket = websocket
         self.access_token = access_token
-        self.repo_dir = repo_dir
-        self.org, self.repo = get_org_and_repo(repo_dir)
+        self.channel_id = channel_id
+        self.channel_mapping = channel_mapping
+        channel_details = channel_mapping[channel_id]
+        self.repo_dir = channel_details['repo_dir']
+
+        self.org, self.repo = get_org_and_repo(self.repo_dir)
+        self.version = None
+        self.rc_hash_url = channel_details['rc_hash_url']
+        self.prod_hash_url = channel_details['prod_hash_url']
+        self.message_count = 0
+
+    def set_repo_details(self, version):
```

Comment:
Can we remove this function and pass in `version` to `release`, `finish_release`, and `wait_for_checkboxes` instead?